core/jobs/batch_jobs/math_interactions_audit_jobs.py

```diff
@@ -33,6 +34,107 @@
 
 (exp_models,) = models.Registry.import_models([models.NAMES.exploration])
 
+TRIVIAL_FUNCTIONS = ['abs', 'sqrt']
+
+
+class Utils:
+    """Utitlity functions for math interaction audit jobs.
+    """
+
+    @staticmethod
+    def contains_math_interactions(model: exp_models.ExplorationModel) -> bool:
+        """Checks if the exploration contains any state with any of the
+        math interactions.
+
+        Args:
+            model: ExplorationModel. Model instance to be checked.
+
+        Returns:
+            bool. Whether the exploration contains math interactions.
+        """
+        return any(
+            state_dict['interaction']['id'] in feconf.MATH_INTERACTION_IDS
+            for state_dict in model.states.values())
+
+    @staticmethod
+    def flat_map_exp_with_states(
+            model: exp_models.ExplorationModel) -> List[Tuple[str, str, dict]]:
+        """Maps exploration model with it's states data.
+
+        Args:
+            model: ExplorationModel. Model instance to be mapped.
+
+        Returns:
+            List[Tuple[str, str, dict]]. List of tuples
+            (exp_id, state_name, state_dict).
+        """
+        return [
+            (model.id, state_name, state_dict)
+            for state_name, state_dict in model.states.items()
+        ]
+
+    @staticmethod
+    def map_with_rule_types(
+            tup: Tuple[str, str, dict]) -> Tuple[str, str, List[str]]:
+        """Maps state tuple with it's rule types.
+
+        Args:
+            tup: Tuple[str, str, dict]. State tuple to be modified.
+
+        Returns:
+            Tuple[str, str, List[str]]. Mapped tuple
+            (exp_id, state_name, list of rules).
+        """
+        answer_groups = tup[2]['interaction']['answer_groups']
+        rule_types = []
+        for answer_group in answer_groups:
+            for rule_spec in answer_group['rule_specs']:
+                rule_types.append(rule_spec['rule_type'])
+
+        return (tup[0], tup[1], rule_types)
+
+    @staticmethod
+    def map_with_rule_inputs(
+            tup: Tuple[str, str, dict]) -> Tuple[str, str, List[str]]:
+        """Maps state tuple with it's rule inputs.
+
+        Args:
+            tup: Tuple[str, str, dict]. State tuple to be modified.
+
+        Returns:
+            Tuple[str, str, List[str]]. Mapped tuple
+            (exp_id, state_name, list of rules).
+        """
+        answer_groups = tup[2]['interaction']['answer_groups']
+        rule_inputs = []
+        for answer_group in answer_groups:
+            for rule_spec in answer_group['rule_specs']:
+                rule_inputs.append(rule_spec['inputs']['x'])
+
+        return (tup[0], tup[1], rule_inputs)
+
+    @staticmethod
+    def uses_non_trivial_functions(
+            tup: Tuple[str, str, List[str]]) -> bool:
+        """Checks if the state contains any rule input that uses functions other
+        than sqrt and abs.
```

Comment:
Don't specify the functions in the docstring if you have them in constant that can be changed later.